### DISCUSSION TITLE
New Features planned for Version 4.2.x

### DISCUSSION THREAD

**Discussion by jones139** [OpenSeizureDetector/Android_Pebble_SD#83] [03_Android App]
This is a list of the features or changes planned for V4.2.x (@aroonpro, please add any others I have missed).

1. Initial support for Android Wear Devices
2. Add support for downloading Machine Learning models as they are developed so we do not need to update the app to change the model.
3. Improve main app screen - move to using Fragments to display different graphs that can be selected by the user rather than showing everything at the same time.
4. Address old (deprecated) code styles (please list the main ones below)

**Comment by AroonPro** [comment]
New test published @aroonPro. New pathway set using manifest encoded broadcastreceiver into wearreceiver.

Startup startupActivity ok.
Reconnect in MainActivity fail.
Link reastablishment  requires at this moment to close all 3 applications.

Still diagnosing other bugs.

Last push also solves couple issues in SdServer.onDestroy

**Comment by lundstrj** [comment]
I just wanted to mention that I appreciate the Exit feature 👍 
I also would like to mention that I miss the Start/Stop server feature. Sometimes it is useful to start OSD just to fiddle with settings and such but without OSD warning me that it cannot find the watch, or that there hasn't been any data received for a while.

Any chance of reintroducing the start/stop feature, or something similar?

**Reply by jones139** [reply]
I'm sure we can - please raise an issue and we'll put it on the list - it might not be in 4.2.x because I'm quite keen to get that pushed to being the production release, so trying to just fix stability bugs at the moment until I'm brave enough to do the switch.

I can't remember why it changed from start/stop to 'exit'.   To avoid the annoying fault pips, I think you can press the 'mute alarms' button and they are suppressed for a while.
